test/integration/integration_test.go

```diff
@@ -0,0 +1,294 @@
+	sdk "github.com/operator-framework/operator-sdk/pkg/sdk"
+	k8sutil "github.com/operator-framework/operator-sdk/pkg/util/k8sutil"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var testConfig *TestConfig
+
+func TestIntegration(t *testing.T) {
+	testConfig = NewTestConfig(t)
+
+	createCRD()
+	defer deleteCRD()
+
+	startOperator()
+
+	// Execute subtests
+	t.Run("TestDNSRecords", testDNSRecords)
+}
+
+func testDNSRecords(t *testing.T) {
+	f := manifests.NewFactory()
+
+	appNamespace, err := f.AppDNSNamespace()
+	if err != nil {
+		t.Fatal(err)
+	}
+	appDeployment, err := f.AppDNSDeployment()
+	if err != nil {
+		t.Fatal(err)
+	}
+	appService, err := f.AppDNSService()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	dnsNamespace, err := f.DNSNamespace()
+	if err != nil {
+		t.Fatal(err)
+	}
+	clusterDNS, err := f.ClusterDNSCustomResource()
+	if err != nil {
+		t.Fatal(err)
+	}
+	clusterDNS.Namespace = testConfig.operatorNamespace
+
+	dnsService, err := f.DNSService(clusterDNS)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cleanup := func() {
+		leftovers := []sdk.Object{
+			clusterDNS,
+			dnsNamespace,
+			appNamespace,
+		}
+		anyFailed := false
+		for _, o := range leftovers {
+			err := sdk.Delete(o)
+			if err != nil && !errors.IsNotFound(err) {
+				t.Logf("failed to clean up object %#v: %s", o, err)
+				anyFailed = true
+			}
+		}
+		if anyFailed {
+			t.Fatalf("failed to clean up resources")
+		}
+	}
+	defer cleanup()
+
+	err = sdk.Create(appNamespace)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = sdk.Create(appDeployment)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = sdk.Create(appService)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = sdk.Create(clusterDNS)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	clusterDomain := getClusterDomain(clusterDNS)
+	serviceIP, servicePort, err := getClusterIP(dnsService)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("configured cluster domain: %s, clusterIP: %s:%s", clusterDomain, serviceIP, servicePort)
+
+	testPod, err := getDNSPod(dnsNamespace.Name)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	type dnsTest struct {
+		domain  string
+		dnsType string
+	}
+	tests := []dnsTest{
+		{
+			domain:  fmt.Sprintf("dns-version.%s.", clusterDomain),
+			dnsType: "txt",
+		},
+		{
+			domain:  fmt.Sprintf("kubernetes.default.svc.%s.", clusterDomain),
+			dnsType: "a",
+		},
+		{
+			domain:  fmt.Sprintf("%s.%s.svc.%s.", appService.Name, appService.Namespace, clusterDomain),
+			dnsType: "a",
+		},
+		{
+			domain:  fmt.Sprintf("_https._tcp.kubernetes.default.svc.%s.", clusterDomain),
+			dnsType: "srv",
+		},
+		{
+			domain:  fmt.Sprintf("_http._tcp.%s.%s.svc.%s.", appService.Name, appService.Namespace, clusterDomain),
+			dnsType: "srv",
+		},
+	}
+
+	for _, test := range tests {
+		digCmd := fmt.Sprintf("dig +noall +answer %s %s @%s -p %s", test.dnsType, test.domain, serviceIP, servicePort)
```

Comment:
net.LookupHost would be more portable but then resolution becomes platform/environment dependent... not sure what the best approach is here. Fine with dig for now, just wanted to start thinking about it